spec/factories/users.rb

```diff
@@ -92,5 +92,17 @@
     trait :invalid_long_username do
       username { Faker::Lorem.characters(number: 30) }
     end
+
+    trait :admin do
+      role { build :role, :admin }
+    end
+
+    trait :redactor do
+      role { build :role, :redactor }
+    end
+
+    trait :reviewer do
+      role { build :role, :reviewer }
+    end
```

Comment:
need rework. Please read FB docs

Reply:
i think that i can use find_or_create_by